arbor/util/rangeutil.hpp

```diff
@@ -82,6 +83,18 @@ void fill(Seq&& seq, const V& value) {
     std::fill(canon.begin(), canon.end(), value);
 }
 
+// Zero a container, specialised for contiguous sequences
+// i.e.: Array, Vector, String.
+
+template <typename Seq,
+          typename = std::enable_if_t<sequence_traits<Seq&&>::is_contiguous>,
+          typename = std::is_trivially_copyable<typename sequence_traits<Seq&&>::value_type>>
```

Comment:
```suggestion
          typename = std::enable_if_t<
              sequence_traits<Seq &&>::is_contiguous &&
              std::is_trivially_copyable_v<typename sequence_traits<Seq &&>>>>
```
The `value_type` of `is_trivially_copyable` is just a type that always exists, so using it this way has no useful effect. You'll want to use `value` or the `_v` definition.
And yes, concepts would be nice!